### PR TITLE
[Design System] DCSO - Create documentation - Checkbox (+ some visual improvements)

### PR DESCRIPTION
**Pull request by levadadenys**
### [Design System] DCSO - Create documentation - Checkbox

* update README.md
* create CHANGELOG.md
* fix rtl support and vertical alignment of the checkbox
* filled in Checkbox component status
* cleanup checkbox storybook
* add different sizes of labels support
* create `apps/src/componentLibrary/common/constants.ts`
* create `apps/src/componentLibrary/common/types.ts`

Before:
Знімок екрана 2023-07-18 о 21 49 34
Знімок екрана 2023-07-18 о 21 49 47
Знімок екрана 2023-07-18 о 21 50 00

After:

Знімок екрана 2023-07-18 о 21 43 42
Знімок екрана 2023-07-18 о 21 43 26
Знімок екрана 2023-07-18 о 21 43 10

## Links
- jira ticket: [DES-300](https://codedotorg.atlassian.net/browse/DES-300)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by levadadenys** [comment]
Waiting for https://github.com/code-dot-org/code-dot-org/pull/52680 to be reviewed and merged.

**Comment by levadadenys** [comment]
Test error is caused y enzyme react shallow renderer not supporting React.memo() (apps/test/unit/templates/MultipleSectionsAssignerTest.js) (https://github.com/enzymejs/enzyme/issues/2196)

File: apps/src/componentLibrary/common/constants.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ */
+
+import {VisualAppearance} from '@cdo/apps/componentLibrary/typography/types';
+
+/**
+ *  This is the map of component size to body text size (visualAppearance)
+ */
+export const componentSizeToBodyTextSizeMap: {[key: string]: VisualAppearance} =
+  {
+    l: 'body-one',
```
Comment (sanchitmalhotra126) [inline apps/src/componentLibrary/common/constants.ts]: If we specifically want to use these keys, we can do something like `type componentSize = 'l' | 'm' | 's' | 'xs'` then declare this as `componentSizeToBodyTextSizeMap: {[key in componentSize]: VisualAppearance}`

**Review by sanchitmalhotra126 — APPROVED** [review]
Looks good! As discussed offline, sounds like we may need to remove the `memo` for now to get tests to pass, but otherwise should be good to go

**Pull request merged**